src/App.tsx

```diff
@@ -1,14 +1,57 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+function loadTodos(
+  setIsLoading: (isLoading: boolean) => void,
+): Promise<Todo[]> {
+  setIsLoading(true);
+
+  return getTodos()
+    .finally(() => {
+      setIsLoading(false);
+    });
+}
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    loadTodos(setIsLoading)
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+      });
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code